### PR TITLE
change driver to support multiple error files

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
change driver to support multiple error files

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/main.bal [hunk 1]
```diff
@@ -61,19 +73,17 @@ public function main(string[] filenames, *Options opts) returns error? {
...
-        return error(err.message());
-    }
-    else {
-        return err;
+function printDiagnostic(err:Diagnostic err) {
+    foreach string line in d:format(err.detail()) {
+        io:fprintln(io:stderr, line);
     }
+    // Don't want details to be displayed twice
+    io:fprintln(io:stderr, err.message());
```
Comment (jclark) [inline compiler/main.bal]: No need to print this.

File: compiler/main.bal [hunk 2]
```diff
@@ -61,19 +73,17 @@ public function main(string[] filenames, *Options opts) returns error? {
             return error(d:messageFormat(`unsupported extension ${ext}`));
         }
     }
+    if errorFileCount != 0 {
+        return error(string `compilation failed due to errors in ${errorFileCount} file(s)`);
```
Comment (jclark) [inline compiler/main.bal]: Don't use `file(s)`: do it properly.

Error message is not quite accurate: should say `compilation of N files failed`.

File: compiler/main.bal [hunk 3]
```diff
@@ -39,17 +39,29 @@ public function main(string[] filenames, *Options opts) returns error? {
         if filenames.length() > 1 {
             return error("multiple input files not supported with --showTypes");
         }
-        check printDiagnostic(showTypes([{ filename: filenames[0] }]));
-        return;
+        CompileError? err = showTypes([{ filename: filenames[0] }]);
+        if err is err:Diagnostic {
+            printDiagnostic(err);
+            return error(""); // we don't need to show any thing for this error because printDiagnostic already does it;
```
Comment (jclark) [inline compiler/main.bal]: No need to change this.

File: compiler/main.bal [hunk 4]
```diff
@@ -39,17 +39,29 @@ public function main(string[] filenames, *Options opts) returns error? {
...
     foreach string filename in filenames {
         var [basename, ext] = basenameExtension(filename);
         if ext == SOURCE_EXTENSION {
-            check printDiagnostic(compileBalFile(filename, basename, check chooseOutputBasename(basename, opts.outDir), nbackOptions, opts));
+            CompileError? err = compileBalFile(filename, basename, check chooseOutputBasename(basename, opts.outDir), nbackOptions, opts);
+            if err is err:Diagnostic {
+                printDiagnostic(err);
+                errorFileCount += 1;
+            }
+            else if err != () {
```
Comment (jclark) [inline compiler/main.bal:59]: Please do the change I suggested in the previous review.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/main.bal (hunk 4 above)
Comment (heshanpadmasiri) [reply compiler/main.bal:59]: Currently throw a compiler error. Created [issue](https://github.com/ballerina-platform/ballerina-lang/issues/34014)

File: compiler/main.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/main.bal]: fixed

File: compiler/main.bal (hunk 2 above)
Comment (heshanpadmasiri) [reply compiler/main.bal]: fixed

File: compiler/main.bal (hunk 3 above)
Comment (heshanpadmasiri) [reply compiler/main.bal]: fixed

**Review by jclark — APPROVED** [review]

**Pull request merged**